tests/python/pants_test/test_base.py

```diff
@@ -698,6 +699,18 @@ def captured_logging(self, level=None):
       root_logger.setLevel(old_level)
       root_logger.removeHandler(handler)
 
+  @contextmanager
+  def warnings_catcher(self):
+    with warnings.catch_warnings(record=True) as w:
+      warnings.simplefilter('always')
+      yield w
```

Comment:
What do you expect to be the use case for non-deprecation test files? Trying to ensure we should add this here and not just inline.

Reply:
It's used in `test_deprecated.py` and `test_options.py`, hence the move from `test_options.py`.